resource_cdap_namespace.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+// https://docs.cdap.io/cdap/current/en/reference-manual/http-restful-api/namespace.html
+func resourceNamespace() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNamespaceCreate,
+		Read:   resourceNamespaceRead,
+		Delete: resourceNamespaceDelete,
+		Exists: resourceNamespaceExists,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceNamespaceCreate(d *schema.ResourceData, m interface{}) error {
+	config := m.(*Config)
+	name := d.Get("name").(string)
+	addr := urlJoin(config.host, "/v3/namespaces", name)
```

Comment:
Maybe pull "/3/namespaces" as a constant and reuse in the 3 functions.

Reply:
I'll look into cleaning up some of the repetition in a follow up. There are a lot of constants littered throughout the code, once we have an established pattern we can clean them up in one PR.